### PR TITLE
Create by year filter

### LINKED ISSUE #204: 6 -  Create filter month/year for medical_shifts and event_procedures
filter by `month/year`.
We can also fix the index method for event_procedures, because I think it only consider month

### PR DESCRIPTION
**Pull request by philipeleandro**
fix #204

### CODE REVIEW

File: app/queries/event_procedures/by_year_query.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  class ByYearQuery < ApplicationQuery
+    attr_reader :year, :relation
+
+    def initialize(year:, relation: EventProcedure)
+      @year = year
+      @relation = relation
+    end
+
+    def call
+      relation.where("EXTRACT(YEAR FROM date) = ?", year)
```
Comment (edimossilva) [inline app/queries/event_procedures/by_year_query.rb:13]: please check if gpt is correct about using `relation.where(date: Date.new(year).all_year)`
Usually we avoid using raw SQL, but I see now problem here

File: app/queries/medical_shifts/by_year_query.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  class ByYearQuery < ApplicationQuery
+    attr_reader :year, :relation
+
+    def initialize(year:, relation: MedicalShift)
+      @year = year
+      @relation = relation
+    end
+
+    def call
+      relation.where("EXTRACT(YEAR FROM start_date) = ?", year)
```
Comment (edimossilva) [inline app/queries/medical_shifts/by_year_query.rb:13]: Same from above

**Review by edimossilva — APPROVED** [review]
🚀

**Review by danielnottingham — APPROVED** [review]

File: app/queries/event_procedures/by_year_query.rb (hunk 1 above)
Comment (philipeleandro) [reply app/queries/event_procedures/by_year_query.rb:13]: Done 01f3a136fe6e8e8869ea0dd2c2a7a6ea78319b53

File: app/queries/medical_shifts/by_year_query.rb (hunk 2 above)
Comment (philipeleandro) [reply app/queries/medical_shifts/by_year_query.rb]: Done 01f3a136fe6e8e8869ea0dd2c2a7a6ea78319b53

Comment (philipeleandro) [reply app/queries/medical_shifts/by_year_query.rb]: done 01f3a136fe6e8e8869ea0dd2c2a7a6ea78319b53

File: app/queries/event_procedures/by_year_query.rb (hunk 1 above)
Comment (philipeleandro) [reply app/queries/event_procedures/by_year_query.rb]: done 8bb46223a5c30a079cede74df662bd95574c9b12

**Pull request merged**